### ISSUE TITLE
[Bug]: `PK column` of Hidden Index Table not updated when PK is a `Composite Primary Key`

### ISSUE THREAD

**Issue by arjunsk** [matrixorigin/matrixone#13674]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main, 1.1-dev, 1.0-dev

### Commit ID

1d0e9850a272bdd1e28bbf41aaed0bf66751e7ec

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

```sql

create database a;
use a;

drop table tbl;
create table tbl(id varchar(20), age varchar(20), embedding vecf32(3), primary key(id, age));
insert into tbl values("1", "10", "[1,2,3]");
insert into tbl values("2", "20", "[1,2,4]");
insert into tbl values("3", "30", "[1,2.4,4]");
insert into tbl values("4", "40", "[1,2,5]");
insert into tbl values("5", "50", "[1,3,5]");
insert into tbl values("6", "60", "[100,44,50]");
insert into tbl values("7", "70", "[120,50,70]");
insert into tbl values("8", "80", "[130,40,90]");
create unique index idx17 using btree on tbl(id);

select * from mo_catalog.mo_indexes where name="idx17";

SELECT * FROM `__mo_index_unique_2f7e0d6e-9fab-11ee-832c-723e89f7b974`;
+--------------------+--------------------+
| __mo_index_idx_col | __mo_index_pri_col |
+--------------------+--------------------+
| 1                  | F1 F10           |
| 2                  | F2 F20           |
| 3                  | F3 F30           |
| 4                  | F4 F40           |
| 5                  | F5 F50           |
| 6                  | F6 F60           |
| 7                  | F7 F70           |
| 8                  | F8 F80           |
+--------------------+--------------------+

update tbl set id="70" where id="7";

SELECT * FROM `__mo_index_unique_a562d91e-9fa2-11ee-8150-723e89f7b974`;
+--------------------+--------------------+
| __mo_index_idx_col | __mo_index_pri_col |
+--------------------+--------------------+
| 1                  | F1 F10           |
| 2                  | F2 F20           |
| 3                  | F3 F30           |
| 4                  | F4 F40           |
| 5                  | F5 F50           |
| 6                  | F6 F60           |
| 8                  | F8 F80           |
| 70                 | F7 F70           |
+--------------------+--------------------+


```

### Expected Behavior

```mysql
SELECT * FROM `__mo_index_unique_2f7e0d6e-9fab-11ee-832c-723e89f7b974`;
+--------------------+--------------------+
| __mo_index_idx_col | __mo_index_pri_col |
+--------------------+--------------------+
| 1                  | F1 F10           |
| 2                  | F2 F20           |
| 3                  | F3 F30           |
| 4                  | F4 F40           |
| 5                  | F5 F50           |
| 6                  | F6 F60           |
| 7                  | F7 F70           |
| 8                  | F8 F80           |
+--------------------+--------------------+

update tbl set id="70" where id="7";

SELECT * FROM `__mo_index_unique_a562d91e-9fa2-11ee-8150-723e89f7b974`;
+--------------------+--------------------+
| __mo_index_idx_col | __mo_index_pri_col |
+--------------------+--------------------+
| 1                  | F1 F10           |
| 2                  | F2 F20           |
| 3                  | F3 F30           |
| 4                  | F4 F40           |
| 5                  | F5 F50           |
| 6                  | F6 F60           |
| 8                  | F8 F80           |
| 70                 | F70 F70           | <----- F70 F70
+--------------------+--------------------+

```

### Steps to Reproduce

```Markdown
The SQL is provided above
```

### Additional information

This issue occurs for Unique Index, Secondary Index and Vector Index

**Comment by arjunsk** [comment]
Another example

```mysql
create table tbl(id varchar(20), age varchar(20), data varchar(30), primary key(id,age));
insert into tbl values("8", "80", "[130,40,90]");
create unique index idx1 using btree on tbl(age);

select * from mo_catalog.mo_indexes where name="idx1";
select * from `__mo_index_unique_297e73c8-a04a-11ee-b3f8-723e89f7b974`;
mysql> select * from `__mo_index_unique_297e73c8-a04a-11ee-b3f8-723e89f7b974`;
+--------------------+--------------------+
| __mo_index_idx_col | __mo_index_pri_col |
+--------------------+--------------------+
| 80                 | F8 F80           |
+--------------------+--------------------+

update tbl set id="newId" where id="8";
select * from `__mo_index_unique_297e73c8-a04a-11ee-b3f8-723e89f7b974`;
+--------------------+--------------------+
| __mo_index_idx_col | __mo_index_pri_col |
+--------------------+--------------------+
| 80                 | F8 F80           |
+--------------------+--------------------+

update tbl set age="newAge" where age="80";
mysql> select * from `__mo_index_unique_297e73c8-a04a-11ee-b3f8-723e89f7b974`;
+--------------------+--------------------+
| __mo_index_idx_col | __mo_index_pri_col |
+--------------------+--------------------+
| newAge             | FnewId F80       |   <--- newId picked from the original table. (should have taken this data from input __mo_cp_key. also __mo_cp_key should be pre-computed)
+--------------------+--------------------+ <--- newAge picked from the current input.

```

**Comment by arjunsk** [comment]
An ad-hoc logic for fixing this issue is here: https://github.com/arjunsk/matrixone/pull/70/files

However, I think there should be a better solution to fix this issue from a stage above it.

**Comment by arjunsk** [comment]
Verified

1. CP keys does not overlap SK
![image](https://github.com/matrixorigin/matrixone/assets/9638314/3eb74763-5572-4524-a6ef-f8a250003a6c)

2. CP keys overlap SK.
![image](https://github.com/matrixorigin/matrixone/assets/9638314/d454692d-4593-48e6-a3bb-c9ddc1cb86de)


We need this change for 1.0-dev as well.

**Comment by qingxinhome** [comment]
The issue has been resolved, please verify and test

**Comment by aressu1985** [comment]
fixed commit:eb63bff0bb47291f6f2d04ed3dc9ea2fcd7654e7
mysql> selectt * from `__mo_index_unique_4bc154a2-a545-11ee-9729-525400777c3a`;
+--------------------+--------------------+
| __mo_index_idx_col | __mo_index_pri_col |
+--------------------+--------------------+
| 1                  | F1 F10           |
| 2                  | F2 F20           |
| 3                  | F3 F30           |
| 4                  | F4 F40           |
| 5                  | F5 F50           |
| 6                  | F6 F60           |
| 7                  | F7 F70           |
| 8                  | F8 F80           |
+--------------------+--------------------+
8 rows in set (0.06 sec)

mysql> update tbl set id="70" where id="7";
Query OK, 1 row affected (0.07 sec)

mysql> select * from `__mo_index_unique_4bc154a2-a545-11ee-9729-525400777c3a`;
+--------------------+--------------------+
| __mo_index_idx_col | __mo_index_pri_col |
+--------------------+--------------------+
| 1                  | F1 F10           |
| 2                  | F2 F20           |
| 3                  | F3 F30           |
| 4                  | F4 F40           |
| 5                  | F5 F50           |
| 6                  | F6 F60           |
| 8                  | F8 F80           |
| 70                 | F70 F70          |
+--------------------+--------------------+
8 rows in set (0.01 sec)
